heaps/min_heap.py

```diff
@@ -19,18 +19,30 @@ def __init__(self):
     def add(self, key, value = None):
         """ This method adds a HeapNode instance to the heap
             If value == None the new node's value should be set to key
-            Time Complexity: ?
-            Space Complexity: ?
+            Time Complexity: O(log n)
+            Space Complexity: O(log n)
         """
-        pass
+        if value == None:
```

Comment:
Small stylistic note: we generally prefer `is` when comparing values to singletons (`True`, `False`, `None`). Occasionally `__eq__` can be overwritten and lead to strange behavior.
```suggestion
        if value is None:
```